### PR TITLE
log the error from scanForKeys

### PR DESCRIPTION
**Pull request by rikonor**
I'm not sure if you were not logging the error intentionally, but it'd be nice if it could be logged.

### CODE REVIEW

**Comment by oliver006** [comment]
Makes sense, thanks for the PR!

File: exporter/redis.go [hunk 1]
```diff
@@ -844,7 +844,7 @@ func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair) (expandedKeys []dbKeyPa
 			}
 			keyNames, err := scanForKeys(c, k.key)
 			if err != nil {
-				log.Errorf("error with SCAN for pattern: %#v", k.key)
+				log.Errorf("error with SCAN for pattern: %#v: %s", k.key, err)
```
Comment (oliver006) [inline exporter/redis.go]: Can you please change this to `"error with SCAN for pattern: %#v err: %s"`

**Comment by rikonor** [comment]
I just realized the same thing happens here: https://github.com/oliver006/redis_exporter/pull/239/files#diff-a75393c3715eaef61ada7312397db7f2R819

Are you OK with me bubbling that error up too?

**Comment by oliver006** [comment]
Yeah, do that while you're at it.

**Comment by oliver006** [comment]
And heads up, I just merged #238 so you need to rebase as i think I changed related code.

**Comment by rikonor** [comment]
OK, rebased, reformatted the log and added the other one as well.

**Comment by rikonor** [comment]
Thanks for the quick merge. We ended up finding out that we had a KEYS call which was killing Redis. Switched that out and all's well now. 😄

**Comment by oliver006** [comment]
Oh interesting. Yeah, KEYS calls returning large sets of keys can be dangerous. Glad to hear you figured it out.

**Pull request merged**